contract/contracts/Tips.sol

```diff
@@ -18,19 +21,40 @@ contract Tips {
     );
     event TipWithdrawal(address to, uint256 amount);
 
-    function pushTip(address[] memory _teamMembersAddresses) positiveAmountRequired external payable {
-        uint256 amountPerAddress = msg.value / _teamMembersAddresses.length;
+    function pushTip(address[] calldata _teamMembersAddresses) positiveAmountRequired external payable {
+        
+        require(_teamMembersAddresses.length > 0, "Must have at least one team member");
+        require(
+            pushLimit >= _teamMembersAddresses.length,
+            "You have too much team members"
+        );
+        uint totalAmount = msg.value + remainder;
+        uint256 amountPerAddress = totalAmount / _teamMembersAddresses.length;
+        
+        // Update the remainder
+        remainder= totalAmount % _teamMembersAddresses.length;
```

Comment:
i still dont understand what is it for, can you explain it @hermannleboss ?

Reply:
i just realize that solidity can have decimal number, still have to think about better solution 🤔